runtime/lib/backend/lightning/lightning_kokkos/LightningKokkosSimulator.cpp

```diff
@@ -29,13 +51,21 @@ auto LightningKokkosSimulator::AllocateQubits(size_t num_qubits) -> std::vector<
         return {};
     }
 
-    const size_t cur_num_qubits = this->device_sv->getNumQubits();
-    const size_t new_num_qubits = cur_num_qubits + num_qubits;
-    this->device_sv = std::make_unique<StateVectorT>(new_num_qubits);
-    return this->qubit_manager.AllocateRange(cur_num_qubits, num_qubits);
+    // at the first call when num_qubits == 0
+    if (!this->GetNumQubits()) {
+        this->device_sv = std::make_unique<StateVectorT>(num_qubits);
+        return this->qubit_manager.AllocateRange(0, num_qubits);
+    }
+
+    std::vector<QubitIdType> result(num_qubits);
+    std::generate_n(result.begin(), num_qubits, [this]() { return AllocateQubit(); });
+    return result;
```

Comment:
Isn't this highly inefficient? I would think the previous method was the right approach and just needed one more instruction to copy the old statevector over (ideally in one batch).

Reply:
this is consistent with LightningSimualtor and we don't call this in the first call to `AllocateQubits` so there shouldn't be any performance concern with the way that qubits allocations/deallocations are currently setup in the end-to-end pipeline. In the future, when we have support for the dynamic qubit management and where we have multiple calls to `AllocateQubits`, we can revisit this support for a batch update of the state-vector.